.github/workflows/check-formatting-llpc.yml

```diff
@@ -0,0 +1,37 @@
+name: Code formatting check
+
+on:
+  pull_request:
+
+jobs:
+  build-and-test:
+    name: clang-format
+    runs-on: "ubuntu-latest"
+    steps:
+      - name: Setup environment
+        run: |
+          sudo apt-get install -yqq clang-format-9
+      - name: Checkout LLPC
+        run: |
+          git clone https://github.com/${GITHUB_REPOSITORY}.git .
+          git fetch origin +${GITHUB_SHA}:${GITHUB_REF} --update-head-ok
+          git checkout ${GITHUB_SHA}
+      - name: Run clang-format
+        run: |
+          git diff ${{ github.base_ref }} -U0 --no-color -- '**/*.cpp' '**/*.cc' '**/*.h' '**/*.hh' \
+            | clang-format-diff-9 -p1 >not-formatted.diff 2>&1
+      - name: Check formatting
+        run: |
+          if ! grep -q '[^[:space:]]' not-formatted.diff ; then
+            echo "Code formatted. Success."
+          else
+            echo "Code not formatted."
+            echo "Run clang-format-diff on your changes and push again."
```

Comment:
Can the script tell the exact command to fix the formatting? That would make it easier to get started.
I think `git diff HEAD@{upstream} -U0 --no-color -- '**/*.cpp' '**/*.cc' '**/*.h' '**/*.hh' | clang-format-diff-9 -p1 -i` should work.

Reply:
Good idea, thanks!